lib/src/err/mod.rs

```diff
@@ -678,7 +680,7 @@ pub enum Error {
 	/// This should be used extremely sporadically, since we lose the type of error as a consequence
 	/// There will be times when it is useful, such as with unusual type conversion errors
 	#[error("Internal database error: {0}")]
-	Internal(String),
+	Internal(&'static str),
```

Comment:
This is a breaking change. In any case, it's not necessarily a better change. It avoids a heap allocation, yes, but a `String` also allows passing in additional context that may only be available at runtime.

Reply:
I was under the impression that changing it to String was a breaking change. Will revert, not touching this any more 😂

Reply:
What do you mean? This is currently a `String` on the [current latest stable version](https://docs.rs/surrealdb/1.1.1/surrealdb/error/enum.Db.html#variant.Internal) and that was the case [going back to v1.0.0](https://docs.rs/surrealdb/1.0.0/surrealdb/error/enum.Db.html#variant.Internal).